backend.go

```diff
@@ -148,3 +160,36 @@ func (b *Backend) Receive() (FrontendMessage, error) {
 	err = msg.Decode(msgBody)
 	return msg, err
 }
+
+// SetAuthType sets the authentication type in the backend.
+// Since multiple message types can start with 'p', SetAuthType allows
+// contextual identification of FrontendMessages. For example, in the
+// PG message flow documentation for PasswordMessage:
+//
+// 		Byte1('p')
+//
+//      Identifies the message as a password response. Note that this is also used for
+//		GSSAPI, SSPI and SASL response messages. The exact message type can be deduced from
+//		the context.
+//
+// Since the Frontend does not know about the state of a backend, it is important
+// to call SetAuthType() after an authentication request is received by the Frontend.
+func (b *Backend) SetAuthType(authType uint32) error {
```

Comment:
Is this only necessary for proxies or like users? I tried pgx with this change and it worked fine, but it'd be good to know for sure.

Reply:
Yes, this is only necessary for proxies. Do you want me to update the comment here to reflect that?